apps/backoffice/src/hooks/use-fetch.ts

```diff
@@ -0,0 +1,173 @@
+ * @returns
+ */
+const manageHttpResponseStatusCode = (
+  status: number
+): HttpResponseClassType => {
+  if (status >= 100 && status <= 199) {
+    return "informational";
+  } else if (status >= 200 && status <= 299) {
+    return "successful";
+  } else if (status >= 300 && status <= 399) {
+    return "redirection";
+  } else if (status >= 400 && status <= 499) {
+    return "clientError";
+  } else if (status >= 500 && status <= 599) {
+    return "serverError";
+  } else {
+    return "unknown";
+  }
+};
+
+const withBearer: WithDefaultsT<"bearerAuth"> = wrappedOperation => params => {
+  return wrappedOperation({
+    ...params,
+    bearerAuth: "VALID_TOKEN" // todo: get session token
+  });
+};
+
+export const clientWithBearerToken: Client<"bearerAuth"> = createClient({
+  baseUrl: API_BACKEND_BASE_URL,
+  fetchApi: (fetch as any) as typeof fetch,
+  withDefaults: withBearer
+});
+
+/** List of all client operations */
+type ClientOperations = typeof clientWithBearerToken;
+
+/** Extract operation request parameters inferred by client operationId */
+type ExtractRequestParams<T extends keyof ClientOperations> = Parameters<
+  ClientOperations[T]
+>[0];
+
+/**
+ * Custom hook for client fetch API.
+ *
+ * This hook use `@pagopa/openapi-codegen-ts` autogenerated client, therefore makes available all and only APIs specified in the project root opeanapi.yaml
+ * @param operationId openapi operationId
+ * @param requestParams request parameters _(as specified in openapi)_
+ * @param responseCodec io-ts expected response codec
+ * @param redirectOnError if true, in case of fetch error, perform an auto redirect to an error page
+ * @param notifyOnError if true, in case of fetch error, show a notification
+ * @returns `data` fetch result of _responseCodec_ type
+ * @returns `error` in case of request error, decode failure or exceptions
+ * @returns `loading` as boolean value for request pending state
+ */
+const useFetch = <RC, T extends keyof ClientOperations>(
+  operationId: T,
+  requestParams: ExtractRequestParams<T>,
+  responseCodec: t.Type<RC>
+  // redirectOnError: boolean = false, // todo: will be implemented soon
+  // notifyOnError: boolean = false // todo: will be implemented soon
+) => {
+  const [data, setData] = useState<RC>();
+  const [error, setError] = useState<UseFetchError>();
+  const [loading, setLoading] = useState(true);
+  const { push } = useRouter();
+
+  /**
+   * SetStateAction\<UseFetchError>
+   * @param status
+   * @param message
+   * @returns
+   */
+  const setUseFetchError = (
+    kind: UseFetchErrorStatusType,
+    message: string,
+    status?: number
+  ) =>
+    setError({
+      kind,
+      message,
+      status
+    });
+
+  const fetchData = async (requestParams: any) => {
+    try {
+      const result = await clientWithBearerToken[operationId](requestParams);
+
+      if (result._tag === "Right") {
+        // Check 401 Unauthorized
+        if (result.right.status === 401) {
```

Comment:
I suggest to assign a variable with a ameaningful name to `result.right`

Reply:
8376da1